src/System.Device.Gpio.Tests/GpioControllerTestBase.cs

```diff
@@ -169,17 +174,18 @@ public void ThrowsIfWritingClosedPin()
         }
 
         [Fact]
-        [Trait("SkipOnTestRun", "Windows_NT")] // Currently, the Windows Driver is defaulting to InputPullDown instead of Input when Closed/Opened.
-        public void OpenPinDefaultsModeToInput()
+        [Trait("SkipOnTestRun", "Windows_NT")] // Currently, the Windows Driver is defaulting to InputPullDown, and it seems this cannot be changed
+        public void OpenPinDefaultsModeToLastMode()
         {
+            // This works for input/output on most systems, but not on pullup/down, since sometimes the hardware doesn't have read possibilities for those (ie. the Raspi 3)
             using (GpioController controller = new GpioController(GetTestNumberingScheme(), GetTestDriver()))
             {
                 controller.OpenPin(OutputPin);
-                Assert.Equal(PinMode.Input, controller.GetPinMode(OutputPin));
                 controller.SetPinMode(OutputPin, PinMode.Output);
                 controller.ClosePin(OutputPin);
                 controller.OpenPin(OutputPin);
-                Assert.Equal(PinMode.Input, controller.GetPinMode(OutputPin));
+                Thread.Sleep(50);
+                Assert.Equal(PinMode.Output, controller.GetPinMode(OutputPin));
```

Comment:
I think the fact that this test is failing now makes sense to me. SysFSDriver wasn't changed, nor did the Controller, but we are changing the expected value in the test, which now makes it fail so that seems reasonable to me. What this tells me is that the sys fs driver on the Pi will always set mode to Output by default, so perhaps what we need to do is to emmulate the desired behavior on the sysfs driver by keeping track of the pinmodes values.

Reply:
That said, I just tried in my Pi and that doesn't seem to be the case if I manually make the change by invoking the file directly.  Can you try adding a larget Sleep just to see if that is what is doing it?